src/FakeItEasy/Core/ArgumentEqualityComparer.cs

```diff
@@ -36,5 +41,55 @@ public bool AreEqual(object expectedValue, object argumentValue)
 
         private IArgumentEqualityComparer? FindHighestPriorityComparer(Type type)
             => this.argumentEqualityComparers.FirstOrDefault(c => c.CanCompare(type));
+
+        private class EnumerableComparer : IArgumentEqualityComparer
+        {
+            private readonly Func<object?, object?, bool> areEqualElements;
+
+            public EnumerableComparer(Func<object?, object?, bool> elementComparer)
+            {
+                this.areEqualElements = elementComparer;
+            }
+
+            public Priority Priority => Priority.Internal;
+
+            public bool CanCompare(Type type) => typeof(IEnumerable).IsAssignableFrom(type);
+
+            public bool AreEqual(object expectedValue, object argumentValue)
+            {
+                if (expectedValue is string expectedString && argumentValue is string argumentString)
+                {
+                    return string.Equals(expectedString, argumentValue);
+                }
+
+                if (argumentValue is not IEnumerable argumentEnumerable)
+                {
+                    return false;
+                }
+
+                var expectedEnumerator = ((IEnumerable)expectedValue).GetEnumerator();
+                var actualEnumerator = argumentEnumerable.GetEnumerator();
+                var anyMoreExpected = expectedEnumerator.MoveNext();
+                var anyMoreActual = actualEnumerator.MoveNext();
+
+                while (anyMoreExpected && anyMoreActual)

```

Comment:
I kind of miss [zip_longest](https://docs.python.org/3/library/itertools.html#itertools.zip_longest).